plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/messages.properties

```diff
@@ -942,5 +943,38 @@ DI_API_INJECTED=error;;;The class being instantiated (or one of its super-types)
 # no parameters required
 DI_ANN_INJECTED_NOT_APPLICABLE=error;;;@Injected annotates a class (abstract or not) defined in an API project.
 
+
+### JSX
+# 0: name of opening JSX element, 1: name of closing JSX element
+JSXELEMENT_OPENING_CLOSING_ELEMENT_NOT_MATCH=error;;;Opening element {0} does not match with closing element {1}.
+# 0: name of the type JSX element is binding to
+REACT_ELEMENT_NOT_FUNCTION_OR_CLASS_ERROR=error;;;JSX element is expected to bind to either a function or class, but bind to type {0} instead.
+# 0: name of the return type of the function that JSX element is binding to
+REACT_ELEMENT_FUNCTION_NOT_REACT_ELEMENT_ERROR=error;;;Expect a function returning React.Element but the return type is {0}.
+# no argument
+REACT_ELEMENT_CLASS_NOT_REACT_ELEMENT_ERROR=error;;;The referred class is not a subtype of React.Component
+# 0: name of the function that JSX element is binding to
+REACT_FUNCTIONAL_COMPONENT_CANNOT_START_WITH_LOWER_CASE=error;;;React functional component {0} cannot start with lower case.
+# 0: name of the class that JSX element is binding to
+REACT_CLASS_COMPONENT_CANNOT_START_WITH_LOWER_CASE=error;;;React class component {0} cannot start with lower case.
+# 0: name of JSX element is binding to
+JSXELEMENT_NOT_BIND_TO_REACT_COMPONENT=error;;;JSX element {0} does not bind to any valid React component.
+# 0: name of JSX element is binding to
+HTMLTAG_UNKNOWN=warning;;;Tag {0} is an unknown HTML tag.
+# 0: name of JSX property
+JSXPROPERTY_ATTRIBUTE_NON_OPTIONAL_PROPERTY_NOT_SPECIFIED=error;;;Non-optional property {0} should be specified.
+# 0: The name of  attribute in spread operator, 1: type of the attribute, 2: the declared type of the corresponding property in "props"
+JSXSPREADATTRIBUTE_WRONG_SUBTYPE=error;;;Attribute {0} has wrong type because {1} not subtype of {2}.
+# 0: The name of attribute in spread operator, 1: name of JSX element
+JSXSPREADATTRIBUTE_NOT_DECLARED_IN_PROPS=warning;;;Attribute {0} is not a declared property in the props of {1}.
+# 0: The name of attribute in JSX property, 1: name of JSX element
+JSXSPROPERTYATTRIBUTE_NOT_DECLARED_IN_PROPS=warning;;;Attribute {0} is not a declared property in the props of {1}.
+# 0: JSX element in non-JSX file
+JSXELEMENT_IN_NON_JSX_RESOURCE=error;;;JSX element is expected to be placed in JSX like resource, was {0}.
+# No parameter
+REACT_NAMESPACE_NOT_ALLOWED=error;;;Namespace to react must be React.
```

Comment:
would be nice to have a common prefix for all N4JSX-related messages
(maybe better be done in a separate second PR)

Reply:
done